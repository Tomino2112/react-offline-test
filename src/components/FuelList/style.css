.fuel-list {
    display: flex;
    margin-bottom: 60px;
    list-style: none;
    text-align: center;
    justify-content: space-between;
    font-size: 1.2em;
}

.fuel-list--major {
    font-size: 2em;
}