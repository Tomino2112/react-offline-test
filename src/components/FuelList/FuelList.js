import React from "react";
import Fuel from "../Fuel/Fuel";
import cs from "classnames";
import "./style.css";

const FuelList = ({ fuels, major = false }) => {
    const className = cs("fuel-list", major && "fuel-list--major");

    return <ul className={className}>
        { fuels.map(fuel => <li key={`card-${fuel.fuel}`}><Fuel name={fuel.fuel} percentage={fuel.perc} /></li>) }
    </ul>
};
 
export default FuelList;