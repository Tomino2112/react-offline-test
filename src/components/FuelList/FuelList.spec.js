import React from "react";
import { shallow } from "enzyme";
import FuelList from "./FuelList";
import mockData from "../../../example_api_response.json";

describe("FuelList", () => {
    test("should render and match snapshot", () => {
        const wrapper = shallow(<FuelList fuels={mockData.data.generationmix} />);

        expect(wrapper).toMatchSnapshot();
    });

    test("should have extra class for major", () => {
        const wrapper = shallow(<FuelList fuels={mockData.data.generationmix} major />);

        expect(wrapper.find('.fuel-list').hasClass('fuel-list--major')).toBeTruthy()
    });
});