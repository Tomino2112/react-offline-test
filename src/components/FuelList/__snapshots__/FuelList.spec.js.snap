// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FuelList should render and match snapshot 1`] = `
<ul
  className="fuel-list"
>
  <li
    key="card-biomass"
  >
    <Fuel
      name="biomass"
      percentage={4.8}
    />
  </li>
  <li
    key="card-coal"
  >
    <Fuel
      name="coal"
      percentage={2.5}
    />
  </li>
  <li
    key="card-imports"
  >
    <Fuel
      name="imports"
      percentage={8.7}
    />
  </li>
  <li
    key="card-gas"
  >
    <Fuel
      name="gas"
      percentage={46.5}
    />
  </li>
  <li
    key="card-nuclear"
  >
    <Fuel
      name="nuclear"
      percentage={16.1}
    />
  </li>
  <li
    key="card-other"
  >
    <Fuel
      name="other"
      percentage={0.3}
    />
  </li>
  <li
    key="card-hydro"
  >
    <Fuel
      name="hydro"
      percentage={0.9}
    />
  </li>
  <li
    key="card-solar"
  >
    <Fuel
      name="solar"
      percentage={14.6}
    />
  </li>
  <li
    key="card-wind"
  >
    <Fuel
      name="wind"
      percentage={5.6}
    />
  </li>
</ul>
`;
